src/x/instrument/options.go

```diff
@@ -63,6 +63,14 @@ func NewOptions() Options {
 	}
 }
 
+// NewTaggedOptions returns a new Options with the metric and logging client
+// tagged with the provided key value.
+func NewTaggedOptions(iOpts Options, k, v string) Options {
+	scope := iOpts.MetricsScope().Tagged(map[string]string{k: v})
+	logger := iOpts.Logger().With(zap.String(k, v))
+	return iOpts.SetMetricsScope(scope).SetLogger(logger)
+}
```

Comment:
nit: Add test? 

Also another nit, perhaps we could add a `NewSubScopeOptions(...)` which performs the same thing but calling SubScope for places that do that?